### DISCUSSION TITLE
3 x 74hc4051 multiplexer and 24 buttons midi

### DISCUSSION THREAD

**Discussion by PanJabuszko** [tttapa/Control-Surface#923] [General]
Hello I want to make a foot-keyboard with 24 buttons - 2 octaves. I have arduino and multiplexers but I don't have any idea how to wire it and run. I have seen the code on github with potentiometers, but can I make it for buttons?

I am counting for your help :DD

**Comment by tttapa** [comment]
The wiring of the multiplexer is the same, regardless of whether you're using potentiometers or switches. You wire the buttons in the same way as if you wired them to an Arduino GPIO pin directly, i.e. between the mulltiplexer pin and ground.

The code for switches is very similar to the code for potentiometers. You simply instantiate some multiplexer objects, and pass the pins of those multiplexers to the `NoteButton` constructor (as opposed to the `CCPotentiometer` constructor in the potentiometer example you referred to).
